### DISCUSSION TITLE
label in epub from tex: issue with validation

### DISCUSSION THREAD

**Discussion by Iacobus1983** [jgm/pandoc#10048] [General]
I have a LaTeX structure like
```
\footnote{foo\label{bar}}
```
Footnote text converts to HTML to 
```
<li id="fn1"><p>foo<span id="bar" label="bar"></span><a href="#fnref1"
class="footnote-back" role="doc-backlink">↩︎</a></p></li>
```
If I generate an epub from my .tex source and I try to validate it with [EPUBCheck](https://github.com/w3c/epubcheck) v4.2.6, I get an error message saying the "label" attribute is not allowed for the element (by the way, the epub works fine).   

Is this `label` within the html actually used for something, or could it be safely removed? Or is it rather an issue of EPUBCheck?

**Comment by jgm** [comment]
Interesting.  Pandoc parses a `\label` as
```
Span ( "bar" , [] , [ ( "label" , "bar" ) ] ) []
```
I would have expected this to be rendered in HTML as `data-label` (normally we add `data-` to attributes that aren't recognized HTML attributes), but I guess `label` is in a list of valid HTML attributes!  (Not for this element perhaps, but we aren't that fine-grained.)

**Reply by jgm** [reply]
The safest thing is probably for us to use `data-label` from the start.

**Reply by jgm** [reply]
Well, I'll go for a more conservative fix and just make sure we add `data-` to label atributes in the HTML writer. This won't affect raw HTML so it should be fine.

**Reply by Iacobus1983** [reply]
@jgm That's wonderful, many thanks!
